spirv_hlsl.cpp

```diff
@@ -2164,6 +2309,134 @@ void CompilerHLSL::emit_texture_size_variants(uint64_t variant_mask, const char
 	}
 }
 
+void CompilerHLSL::analyze_meshlet_writes()
+{
+	if (get_execution_model() == ExecutionModelMeshEXT)
+	{
+		uint32_t id_per_vertex = 0;
+		uint32_t id_per_primitive = 0;
+		uint32_t id_payload = 0;
+		ir.for_each_typed_id<SPIRVariable>([&](uint32_t id, SPIRVariable &var) {
+			auto &type = this->get<SPIRType>(var.basetype);
+			bool block = has_decoration(type.self, DecorationBlock);
+			if (var.storage == StorageClassOutput && block && is_builtin_variable(var))
+			{
+				auto flags = get_buffer_block_flags(var.self);
+				if (flags.get(DecorationPerPrimitiveEXT))
+					id_per_primitive = var.self;
```

Comment:
The assumption here is that there is one unique id_per_primitive and id_per_vertex. Is this assumption correct?

Reply:
For per-vertex - valid, meshlet must have at least `gl_Position`
For per-primitive - fixed, but emitting empty `gl_MeshPerPrimitiveEXT` struct in spir-v